llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -804,7 +804,8 @@ void ConstraintInfo::transferToOtherSystem(
     }
     break;
   case CmpInst::ICMP_SLT:
-    if (doesHold(CmpInst::ICMP_SGE, A, ConstantInt::get(B->getType(), 0)))
+    if (doesHold(CmpInst::ICMP_SGE, A, ConstantInt::get(B->getType(), 0)) ||
+        isKnownNonNegative(A, DL, /*Depth=*/MaxAnalysisRecursionDepth - 1))
```

Comment:
There are a bunch of places in this code doing the doesHold() check with SGE 0. Any particular reason why only this one in particular should be adjusted?

Reply:
Just had to add test cases for all the others. Should be done in 97a6c1534f6 and updated the patch